api/src/main/java/com/netflix/iceberg/expressions/Expressions.java

```diff
@@ -96,6 +96,10 @@ public static Expression not(Expression child) {
     return new UnboundPredicate<>(Expression.Operation.NOT_EQ, ref(name), value);
   }
 
+  public static <T> UnboundPredicate<T> startWith(String name, T value) {
```

Comment:
Why does this accept `T` and not `String`? Shouldn't this always be a string, or does this also allow startsWith evaluation on byte arrays?

Reply:
Also, is it a typo that this is `startWith` and not `startsWith`?

Reply:
If I leave this as String it won't compile, I wanted to have it throw `ValidationException` as the other operations
you can see the test [here](https://github.com/Netflix/iceberg/pull/78/files/072faaa7f16d6a9b07aec7925ff1221f3d1478f5#diff-9f0324576b3dbe3551b3756356cb660dR169)
if I change it to string it won't compile

Reply:
What is the compile error? I'm surprised it won't compile if you remove the type variable and use String.

Reply:
I had a test checking that if I try to run startWith on anything other then string it will throw `ValidationError` which is the exception I throw in the `bind` method.
if `startWith` excepting only strings  test would not compile.
I can remove this and change the type to String.
but does it mean that the `bind` method is useless for `startWith`?

Reply:
You can always use the `Expressions.predicate` method to create the predicate with a non-string value. I think that it's better to make `Expressions.startsWith` accept only a String so that the type system helps make expressions correct, not just binding checks.